commit 49ab56ac6e1b907b7dadb72a4012460359feaf0e
Author: Alok Kataria <akataria@vmware.com>
Date:   Sat Nov 1 18:34:37 2008 -0700

    x86: add X86_FEATURE_HYPERVISOR feature bit
    
    Impact: Number declaration only.
    
    Add X86_FEATURE_HYPERVISOR bit (CPUID level 1, ECX, bit 31).
    
    Signed-off-by: H. Peter Anvin <hpa@zytor.com>

Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN a/include/asm-x86/cpufeature.h b/include/asm-x86/cpufeature.h
--- a/include/asm-x86/cpufeature.h	2009-07-16 14:25:56.000000000 -0600
+++ b/include/asm-x86/cpufeature.h	2009-07-16 14:28:37.000000000 -0600
@@ -91,6 +91,7 @@
 #define X86_FEATURE_CX16	(4*32+13) /* CMPXCHG16B */
 #define X86_FEATURE_XTPR	(4*32+14) /* Send Task Priority Messages */
 #define X86_FEATURE_DCA		(4*32+18) /* Direct Cache Access */
+#define X86_FEATURE_HYPERVISOR	(4*32+31) /* Running on a hypervisor */
 
 /* VIA/Cyrix/Centaur-defined CPU features, CPUID level 0xC0000001, word 5 */
 #define X86_FEATURE_XSTORE	(5*32+ 2) /* on-CPU RNG present (xstore insn) */
@@ -188,6 +189,7 @@ extern const char * const x86_power_flag
 #define cpu_has_gbpages		boot_cpu_has(X86_FEATURE_GBPAGES)
 #define cpu_has_arch_perfmon	boot_cpu_has(X86_FEATURE_ARCH_PERFMON)
 #define cpu_has_pat		boot_cpu_has(X86_FEATURE_PAT)
+#define cpu_has_hypervisor	boot_cpu_has(X86_FEATURE_HYPERVISOR)
 
 #if defined(CONFIG_X86_INVLPG) || defined(CONFIG_X86_64)
 # define cpu_has_invlpg		1
